.github/workflows/build.yml

```diff
@@ -3,29 +3,32 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 #
-name: Go
+name: build
 on: [push, pull_request]
 jobs:
 
   unitTest:
-    name: Unit Test
+    name: Unit test
     runs-on: ${{ matrix.os }}
+    timeout-minutes: 5
     strategy:
       matrix:
         os: [ubuntu-latest, macOS-latest]
     steps:
 
-    - name: Set up Go 1.13
+    - name: Setup Go 1.13
       uses: actions/setup-go@v1
       with:
         go-version: 1.13
       id: go
 
     - uses: actions/checkout@v1
 
-    - name: Unit Test
-      timeout-minutes: 5
+    - name: Run unit test
       run: make unit-test
+    
+    - name: Upload coverage to Codecov  
+      uses: codecov/codecov-action@v1.0.2
```

Comment:
Add `timeout-minutes: 5` per job
and move it to `Run checks` job bc here we have matrix strategy and it (Codecov) doesn't work for mac

Reply:
codecov uses the result of unit test run. I had to duplicate the unit test section.

Reply:
(I left the matrix so the OS is filled into the check name).

Reply:
replaced with a conditional.